ruby/lib/asciidoctor/extensions/asciidoctor_kroki/extension.rb

```diff
@@ -74,6 +75,14 @@ def process(processor, parent, attrs, diagram_type, diagram_text)
 
       private
 
+      def prepend_plantuml_config(diagram_text, diagram_type, doc)
+        if diagram_type == :plantuml && doc.attributes['kroki-plantuml-include']
+          config = File.read(doc.attributes['kroki-plantuml-include'])
```

Comment:
I think we should use `attr` and `attr?` methods:

```suggestion
        if diagram_type == :plantuml && doc.attr?('kroki-plantuml-include')
          config = File.read(doc.attr('kroki-plantuml-include'))
```

I'm not a big fan of assigning a value in a condition but it could be reduced to:


```rb
if diagram_type == :plantuml && (plantuml_include = doc.attr('kroki-plantuml-include'))
  File.read(plantuml_include) + '\n' + diagram_text
else
  diagram_text
end
```

Reply:
I'm a bit concerned that we have a different implementation in Ruby but at the same time we won't implement the PlantUML processor in Ruby in this pull request.

However, could you please add a "TODO" to make it clear that we need to implement the PlantUML preprocessor in Ruby and replace `File.read(plantuml_include)` by `"!include #{plantuml_include}"`?

Reply:
I don't think we should rely on "!include" before we have a preprocessor for ruby.
Wouldn't that limit the ruby implementation to files on the server and URLs, as the content of local files won't be sent to the server?

Reply:
No worries, you are 100% correct we should not rely on `!include` until the preprocessor is implemented in Ruby but the "TODO" will help to remember this decision and what needs to be done 👌🏻